game/kernel/kscheme.cpp

```diff
@@ -318,24 +318,47 @@ u64 make_string_from_c(const char* c_str) {
  * creates a function object on the global heap.
  *
  * The implementation is to create a simple trampoline function which jumps to the C function.
+ *
+ *
  */
 Ptr<Function> make_function_from_c(void* func) {
   // allocate a function object on the global heap
   auto mem = Ptr<u8>(
-      alloc_heap_object(s7.offset + FIX_SYM_GLOBAL_HEAP, *(s7 + FIX_SYM_FUNCTION_TYPE), 0x40));
+      alloc_heap_object(s7.offset + FIX_SYM_GLOBAL_HEAP, *(s7 + FIX_SYM_FUNCTION_TYPE), 0x80));
```

Comment:
we did a lot of debugging in this file, not sure how much of this we want to keep